model/minsky.h

```diff
@@ -297,6 +298,8 @@ namespace minsky
     bool running=false; ///< controls whether simulation is running
     bool reverse=false; ///< reverse direction of simulation
     void reset(); ///<resets the variables back to their initial values
+    /// @return optimize() returns an integer of the status of non-linear least square optimization, either GSL_SUCCESS, GSL_EMAXITER or GSL_ENOPROG
+    int optimize(double xf[],OptParams& optPars); ///<minimize residual function by LM optimization    
```

Comment:
optimize may as well be a private method, as there is no way you can use it without knowing the structure of OptParams.

Reply:
Agreed, it will have to go into a pimpl pattern like RKdata and I don't think it can be merged at all as it currently stands. At the very minimum, a tick box in the Runge Kutta drop-down menu in Minsky's GUI will be needed as an option to activate optimization. Most likely, it will require its own drop-down menu.